static/js/components/VideoPlayer.js

```diff
@@ -129,38 +145,66 @@ export default class VideoPlayer extends React.Component {
     this.videoNode.style.left = `${left}px`;
     this.videoNode.style.top = `${top}px`;
     // $FlowFixMe prop does not have to be a number
-    this.videoNode.style[prop] = 'scale(2)';
+    this.videoNode.style[prop] = "scale(2)";
     this.configureCameras();
 
   }
 
   toggleFullscreen() {
-    if (isFullscreen()) {
+    let fullscreen = isFullscreen();
+    if (fullscreen) {
       // $FlowFixMe
       document[FULLSCREEN_API.exitFullscreen]();
     } else {
       // $FlowFixMe videoContainer.parentElement is not going to be null
       this.videoContainer.parentElement[FULLSCREEN_API.requestFullscreen]();
     }
+    this.player.el_.dispatchEvent(new Event(`fullscreen ${fullscreen ? "off" : "on"}`));
+  }
+
+  analyzeEvent(action: string, label: string) {
+    let percentTracked = this.percentTracked;
+    this.player.on(action, function() {
+      if (action === "timeupdate") {
+        // Track percentage played in increments of 10
+        const currentTime = Math.round(this.currentTime());
+        const duration = Math.round(this.duration());
+        let percentPlayed = Math.round(currentTime / duration * 100);
+        _.range(10,100,10).forEach((percent:number) => {
+          if (percentPlayed >= percent && !percentTracked.includes(percent)) {
+            sendGAEvent("video", action, "percentPlayed", percent);
+            percentTracked.push(percent);
+          }
+        });
+      } else {
+        sendGAEvent("video", `${action}`, label, Math.round(this.currentTime()));
+      }
+    }, false);
   }
 
   componentDidMount() {
     const { video } = this.props;
     const cropVideo = this.cropVideo.bind(this);
+    const analyzeEvent = this.analyzeEvent.bind(this);
```

Comment:
if you declare `this.analyzeEvent` as an arrow function I think you can avoid doing this `.bind(this)` fiddling here

Reply:
Thanks, will do